Move data stream transport and rest action to xpack

martijnvg opened the pull request:
* Actions are moved to xpack core.
* Transport and rest actions are moved the data-streams module.
* Removed data streams methods from Client interface.
* Adjusted tests to use client.execute(...) instead of data stream specific methods.

martijnvg committed:
Move data stream transport and rest action to xpack

* Actions are moved to xpack core.
* Transport and rest actions are moved the data-streams module.
* Removed data streams methods from Client interface.
* Adjusted tests to use client.execute(...) instead of data stream specific methods.

elasticmachine commented:
Pinging @elastic/es-core-features (:Core/Features/Data streams)

martijnvg committed:
only attempt to delete all data streams if xpack is installed

andreidan reviewed (approved):
LGTM

danhermann commented on x-pack/plugin/data-streams/src/test/java/org/elasticsearch/xpack/datastreams/action/DeleteDataStreamTransportActionTests.java:
```diff
@@ -0,0 +1,208 @@
+        return s;
+    }
+
+    /**
+     * Constructs {@code ClusterState} with the specified data streams and indices.
+     *
+     * @param dataStreams The names of the data streams to create with their respective number of backing indices
+     * @param indexNames  The names of indices to create that do not back any data streams
+     */
+    public static ClusterState getClusterStateWithDataStreams(List<Tuple<String, Integer>> dataStreams, List<String> indexNames) {
```
I suggest moving this method to DataStreamTestHelper since it's used in a lot of tests in the server module.

martijnvg replied:
Good point. This is why ci/1 failed. I've addressed this.

danhermann reviewed (approved):
LGTM. Left a minor comment about a test helper method.

martijnvg committed:
moved test util method to DataStreamTestHelper

martijnvg committed:
fixed checkstyle violation

dakrone commented on x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/rest/RestDeleteDataStreamAction.java:
```diff
@@ -1,3 +1,9 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
 /*
  * Licensed to Elasticsearch under one or more contributor
  * license agreements. See the NOTICE file distributed with
```
This license needs to be removed

martijnvg replied:
whoops 😬 
dual licensing was not the intend of this change :)

dakrone commented on x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/rest/RestCreateDataStreamAction.java:
```diff
@@ -1,3 +1,9 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
 /*
  * Licensed to Elasticsearch under one or more contributor
  * license agreements. See the NOTICE file distributed with
```
This license needs to be removed

dakrone commented on x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/rest/RestGetDataStreamsAction.java:
```diff
@@ -1,3 +1,9 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
 /*
  * Licensed to Elasticsearch under one or more contributor
  * license agreements. See the NOTICE file distributed with
```
Same here

dakrone reviewed (approved):
LGTM, thanks Martijn. I left a couple of comments, it looks like the licenses were doubled up, so we need to remove the lower license and leave the xpack one for these files

martijnvg committed:
Now that ds apis are in xpack and ESIntegTestCase
no longers deletes all ds, do that in the MlNativeIntegTestCase
class for ml tests.

martijnvg committed:
removed forgotten apache2 license headers

martijnvg committed:
Merge remote-tracking branch 'es/master' into move_ds_action_ds_module

martijnvg committed:
fix issues after merging in master

martijnvg committed:
Merge remote-tracking branch 'es/master' into move_ds_action_ds_module

martijnvg committed:
fixed test compile error
